src/lib.rs

```diff
@@ -69,15 +69,96 @@
+
+/// A struct that represents styles that should be used by `color_spantrace`
+#[derive(Debug, Copy, Clone, Default)]
+pub struct Styles {
+    file: Style,
+    line_number: Style,
+    target: Style,
+    fields: Style,
+    active_line: Style,
+}
+
+impl Styles {
+    /// Create blank styles
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Styles for a dark background. This is the default
+    pub fn dark() -> Self {
+        Self {
+            file:         style().purple(),
+            line_number:  style().purple(),
+            active_line:  style().white().bold(),
+
+            // XXX your trait from this file (which I have removed now) produced "xterm" colors. `owo_colors` documents the `xterm` module as follows: "XTerm 256-bit colors. Not as widely supported as standard ANSI but contains 240 more colors." (see *). Maybe it would be best to switch to something else now. In this case, the colors also have to be changed in `owo_colors`
+            // * https://github.com/jam1garner/owo-colors/blob/c0db9127cbc355dec36fbfa45b9adc534b72c4ad/src/colors.rs#L173
+            target:       style().color(XtermColors::UserBrightRed),
+            fields:       style().color(XtermColors::UserBrightCyan),
+        }
+    }
+
+    // XXX same as with `light` in `color_eyre`
+    /// Styles for a light background
+    pub fn light() -> Self {
+        Self {
+            file:         style().purple(),
+            line_number:  style().purple(),
+            target:       style().red(),
+            fields:       style().blue(),
+            active_line:  style().bold(),
+        }
+    }
+
+    /// Styles printed paths
+    pub fn file(mut self, style: Style) -> Self {
+        self.file = style;
+        self
+    }
+
+    /// Styles the line number of a file
+    pub fn line_number(mut self, style: Style) -> Self {
+        self.line_number = style;
+        self
+    }
+
+    /// Styles the target (i.e. the module and function name, and so on)
+    pub fn target(mut self, style: Style) -> Self {
+        self.target = style;
+        self
+    }
+
+    // XXX is this correct?
+    /// Styles fields of the `tracing` crate (in the context of `color_spantrace`, the arguments of functions and methods)
+    pub fn fields(mut self, style: Style) -> Self {
+        self.fields = style;
+        self
+    }
+
+    /// Styles the selected line of displayed code
+    pub fn active_line(mut self, style: Style) -> Self {
+        self.active_line = style;
+        self
+    }
+}
+
+/// Sets the global styles. This can only be set once and otherwise fails (`Err` contains the `Styles` that was passed to `set_styles`). Note: `colorize` sets the global styles implicitly, if they were not set already. So calling `colorize` and then `set_styles` fails
```

Comment:
I don't think there's any advantage to us passing the `Styles` back out because at that point it is no longer useful for anything afaik. Let's add an Error type to just explain what went wrong.
```suggestion
/// Sets the global styles.
///
/// # Details
///
/// This can only be set once and otherwise fails.
///
/// **Note:** `colorize` sets the global styles implicitly, if they were not set already. So calling `colorize` and then `set_styles` fails
```